ffi/diplomat/Cargo.toml

```diff
@@ -93,6 +93,20 @@ icu_properties = ["dep:icu_properties", "dep:icu_collections", "dep:unicode-bidi
 icu_segmenter = ["dep:icu_segmenter"]
 icu_timezone = ["dep:icu_timezone", "dep:icu_calendar"]
 
+compiled_data = [
+    "icu_calendar?/compiled_data",
+    "icu_collator?/compiled_data",
+    "icu_datetime?/compiled_data",
+    "icu_decimal?/compiled_data",
+    "icu_displaynames?/compiled_data",
+    "icu_list?/compiled_data",
+    "icu_locid_transform?/compiled_data",
+    "icu_normalizer?/compiled_data",
+    "icu_plurals?/compiled_data",
+    "icu_properties?/compiled_data",
+    "icu_segmenter?/compiled_data",
+    "icu_timezone?/compiled_data",
+]
 
```

Comment:
Issue: You should put `default-features = false` on all the component crates if we want to clean up the dependency tree for icu_capi, which we absolutely do